src/caffe/layers/accuracy_layer.cpp

```diff
@@ -50,11 +54,14 @@ void AccuracyLayer<Dtype>::Forward_cpu(const vector<Blob<Dtype>*>& bottom,
   const int num_labels = bottom[0]->shape(label_axis_);
   vector<Dtype> maxval(top_k_+1);
   vector<int> max_id(top_k_+1);
+  vector<Dtype> accuracies(dim, 0);
+  vector<Dtype> nums(dim, 0);
```

Comment:
My opinion is to keep these two variables as computation buffers and put them as member variable blobs as is done in many other layers (actually you only need to create one such vector and use top[1] itself as another). The advantages are
* you don't need to reallocate them every time.
* you don't need to allocate them if `top.size() == 1`

>They are usually small and it doesn't cost much to create them.

Not necessarily true, sometimes we may want to do 10,000+ category classification in caffe.